### PR TITLE
Fr 1.6

### PR DESCRIPTION
**Pull request by RyanCheungJF**
Things done:

- [X] Add linkage between sign up and login
- [X] Refactor code base to standardize some formatting
- [X] Dummy profile page to view info of user in the future
- [x] Change password backend + DB changes

To test:
Open up dev console, login with a normal user.
When changing password, check console for feedback on whether it works.
Re-login with new password.

### CODE REVIEW

File: user-service/controller/user-controller.js [hunk 1]
```diff
@@ -67,28 +67,50 @@ export const deleteUser = async (req, res) => {
...
   return res.status(401).send('Invalid login credentials')
 }
 
+export const updateUserPassword = async (req, res) => {
+  try {
+    const { username, newPassword } = req.body
+    if (!(username && newPassword)) {
+      return res
+        .status(400)
+        .json({ message: 'Username and/or Password are missing!' })
```
Comment (Silvernitro) [inline user-service/controller/user-controller.js:94]: Just a note to self here: maybe we can consider implementing a query validator middleware next time to extract the boilerplate logic here that's used in every request handler.

**Review by Silvernitro — COMMENTED** [review]
The code LGTM! but I couldn't get it to work while testing tho... I was trying to change the password of my account (`_id: 631e9ec5a9d1353c6ad12ab1`) but received 400. I changed the hardcoded id in `ProfilePage.js` to my account id tho!

File: frontend/src/pages/ProfilePage.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const handleChangePassword = async () => {
+    if (!newPassword) {
+      // will add in some ui component in the future to reflect
+      // but for now can just check with console.log
+      console.log('Password cannot be empty!')
+      return
+    }
+    try {
+      const res = await changeUserPassword(uuid, newPassword)
+      console.log('success ', res)
```
Comment (dannytayjy) [inline frontend/src/pages/ProfilePage.js:24]: I think u should add a navigate back to home page here after password change is successful

File: frontend/src/pages/LoginPage.js [hunk 3]
```diff
@@ -13,8 +17,9 @@ const LoginPage = () => {
 
   const handleLogin = async (username, password) => {
     setIsLoginSuccess(false)
+    checkFormFields(username, password)
     try {
-      const res = await loginUser(username, password)
+      const res = await loginUser(username, password, setErrorDialog)
```
[the remark this answers is not in the document]
Comment (RyanCheungJF) [reply frontend/src/pages/LoginPage.js:22]: Yup, good spot! Thanks and will update!

File: user-service/controller/user-controller.js (hunk 1 above)
Comment (RyanCheungJF) [reply user-service/controller/user-controller.js:94]: Sounds good, we can refactor that in 👍🏻

**Review by Silvernitro — APPROVED** [review]
LGTM and tested to be working!

**Pull request merged**